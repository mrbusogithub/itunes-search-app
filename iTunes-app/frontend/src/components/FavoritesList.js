import React from "react";
import { useDispatch, useSelector } from "react-redux"; // Import useDispatch and useSelector hooks from React Redux
import { removeFromFavorites } from "../store/actions/favoritesActions"; // Import removeFromFavorites and setFavorites actions from the favoritesActions file

import ListItem from "./ListItem"; // Import the ListItem component

// FavoritesList component displays a list of favorite items
const FavoritesList = () => {
  const dispatch = useDispatch();
  const favorites = useSelector((state) => state.favorites);

  // Function to handle removal of an item from favorites
  const handleRemove = (index) => {
    dispatch(removeFromFavorites(index));
  };

  return (
    <div>
      <ul>
        {/* Map through favorites to display each item using ListItem component */}
        {favorites.map((favorite, index) => (
          <ListItem
            key={index}
            item={favorite}
            onButtonClick={() => handleRemove(index)}
            buttonText="Remove"
          />
        ))}
      </ul>
    </div>
  );
};

export default FavoritesList;
