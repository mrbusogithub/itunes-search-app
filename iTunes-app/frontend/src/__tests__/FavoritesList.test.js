import React from "react"; // Import React library for creating React components
import renderer from "react-test-renderer"; // Import renderer from react-test-renderer to render React components for testing
import FavoritesList from "../components/FavoritesList"; // Import the FavoritesList component to be tested
import configureStore from "redux-mock-store"; // Import configureStore from redux-mock-store to create a mock Redux store for testing
import { Provider } from "react-redux"; // Import Provider from react-redux to wrap the tested component with a Redux store

// Create a mock Redux store for testing
const mockStore = configureStore([]);

// Test case to ensure that FavoritesList renders correctly
test("FavoritesList renders correctly", () => {
  // Mock initial state of the Redux store with favorite albums
  const store = mockStore({
    favorites: [
      {
        albumName: "Album 1",
        artistName: "Artist 1",
        releaseDate: "2022-01-01",
      },
      {
        albumName: "Album 2",
        artistName: "Artist 2",
        releaseDate: "2022-02-01",
      },
    ],
  });

  // Render the FavoritesList component with the mock Redux store
  const component = renderer.create(
    <Provider store={store}>
      <FavoritesList />
    </Provider>
  );

  // Convert the rendered component to a JSON tree
  const tree = component.toJSON();

  // Compare the rendered tree with the stored snapshot
  expect(tree).toMatchSnapshot();
});
