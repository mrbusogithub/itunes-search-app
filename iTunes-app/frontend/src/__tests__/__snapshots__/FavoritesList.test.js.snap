// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesList renders correctly 1`] = `
<div>
  <ul>
    <li
      className="list-group-item d-flex justify-content-between align-items-center"
      style={
        Object {
          "backgroundColor": "lightblue",
          "borderRadius": "5px",
        }
      }
    >
      <img
        alt="Album Cover"
        className="mr-3 rounded ms-3"
      />
      <div
        className="content"
        style={
          Object {
            "color": "dimgray",
          }
        }
      >
        <p>
          Album 1
        </p>
        <p>
          Artist 1
        </p>
        <p>
          2022-01-01
        </p>
      </div>
      <button
        className="btn btn-primary btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Remove
      </button>
    </li>
    <li
      className="list-group-item d-flex justify-content-between align-items-center"
      style={
        Object {
          "backgroundColor": "lightblue",
          "borderRadius": "5px",
        }
      }
    >
      <img
        alt="Album Cover"
        className="mr-3 rounded ms-3"
      />
      <div
        className="content"
        style={
          Object {
            "color": "dimgray",
          }
        }
      >
        <p>
          Album 2
        </p>
        <p>
          Artist 2
        </p>
        <p>
          2022-02-01
        </p>
      </div>
      <button
        className="btn btn-primary btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Remove
      </button>
    </li>
  </ul>
</div>
`;
